debian/upstart

```diff
@@ -0,0 +1,33 @@
+description "drserv is an authenticated HTTP API for publishing .deb package to a repo"
+
+start on runlevel [2345]
+stop on runlevel [!2345]
+
+# never stop trying to respawn
+respawn
+respawn limit unlimited
+
+# Drop privileges
+setuid drserv
+setgid nogroup
+
+env DEFAULTFILE=/etc/default/drserv
+
+pre-start script
+  [ -f $DEFAULTFILE ] && . $DEFAULTFILE
+
+  # don't start unless explicitly enabled in config
+  if [ -z "$ENABLED" ] ; then
```

Comment:
The current best practice for these things are, as I understand them, to do `ENABLED=true` (or `false` and then have the `if` clause look like `[[ -n "$ENABLED" ]] && $ENABLED`. That would catch if it is not set, but it can still explicitly be `true` or `false`.

Also, using `[[ ... ]]` over `[ ... ]` is also a best practice since `[` will start a new process and `[[` is the same thing but internal to the shell.